jni/src/main/java/com/horizen/schnorrnative/SchnorrKeyPair.java

```diff
@@ -21,6 +21,12 @@ public SchnorrKeyPair(SchnorrSecretKey secretKey) {
         this.publicKey = secretKey.getPublicKey();
     }
 
+    private static native SchnorrKeyPair nativeDeriveFromSeed(byte[] seed);
+
+    public static SchnorrKeyPair deriveFromSeed(byte[] seed) {
```

Comment:
Minor thing: We can just overload the generate() function by passing the additional seed parameter instead of using a different name

Reply:
That would require an API breaking change. If we are okay with this I can make that change.

Reply:
Why ? Just add a separate generate(byte[] seed) function

Reply:
Ah right because Rust side we don't have the overload. You could manage it by adding a single native function that always take a seed, and let the `generate(void)` call that one passing an empty seed (that you already handled Rust side like a normal generate without seed). Btw is not so important